rasa/core/channels/slack.py

```diff
@@ -23,7 +25,9 @@ def name(cls) -> Text:
     def __init__(self, token: Text, slack_channel: Optional[Text] = None) -> None:
 
         self.slack_channel = slack_channel
-        self.client = WebClient(token, run_async=True)
+        self.client = WebClient(
+            token, run_async=True, session=ClientSession(trust_env=True)
```

Comment:
@BrainConverter I'm a bit torn at the moment. By creating the `session` manually, we loose access to slacks default configuration (namely the configured timeout). I discovered that `WebClient` also accepts a `proxy` argument which you could use to pass the proxy URL. 
In addition I think it could be preferable for users to configure the proxy URL via the `credentials.yml` as all config for the channel is in one place then. Hence, I think it be slightly preferable to set the `proxy` arg via `credentials.yml`. What do you think?

Reply:
You are right this might not be the best implementation.
I'll think about another solution and put this PR back to draft.

Reply:
Please feel free to tag me and discuss ideas / opinions with me.

Reply:
Since the functionality is already implemented in Rasa 2.x is there a best practice how to sync the different versions of the Slack connector (Rasa 2.x and 1.x) ?